### PR TITLE
Add Examples for export suffix hint feature in GDScript and C#

### PR DESCRIPTION
**Pull request by ShawnHardern**
In response to the discussion in [godot-proposals #10498](https://github.com/godotengine/godot-proposals/issues/10498), this pull request updates the documentation to surface this functionality

### CODE REVIEW

**Review by 0stam — CHANGES_REQUESTED** [review]
You should add dots at the ends of paragraphs for consistency with the rest of docs. Other than that looks great, I personally didn't know the unit hints exist ;)

According to the workflow you should also [squash the commits](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase) before merging.

**Comment by ShawnHardern** [comment]
> You should add dots at the ends of paragraphs for consistency with the rest of docs. Other than that looks great, I personally didn't know the unit hints exist ;)
> 
> According to the workflow you should also [squash the commits](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase) before merging.

@0stam Good spot! thanks I have pushed an update, will squash the commits as well !

File: tutorials/scripting/c_sharp/c_sharp_exports.rst
Comment (ShawnHardern) [reply tutorials/scripting/c_sharp/c_sharp_exports.rst]: Updated 👍🏻

File: tutorials/scripting/gdscript/gdscript_exports.rst
Comment (ShawnHardern) [reply tutorials/scripting/gdscript/gdscript_exports.rst]: Updated 👍🏻

File: tutorials/scripting/gdscript/gdscript_exports.rst
Comment (tetrapod00) [inline tutorials/scripting/gdscript/gdscript_exports.rst:188]:
```suggestion
    @export_custom(PROPERTY_HINT_NONE, "suffix:m/s\u00b2") var gravity: float
```

The original line you had compiles, but is not editable. 
Quick testing:
```
@export_custom(PROPERTY_HINT_NONE, "suffix:m/s\u00b2") var invalid
@export_custom(PROPERTY_HINT_NONE, "suffix:m/s\u00b2") var valid_0 : float
@export_custom(PROPERTY_HINT_NONE, "suffix:m/s\u00b2") var valid_1 = 0.0
@export_custom(PROPERTY_HINT_NONE, "suffix:m/s\u00b2") var valid_2 := 0.0

@export_custom(PROPERTY_HINT_NONE, "suffix:m/s\u00b2") var invalid_int
@export_custom(PROPERTY_HINT_NONE, "suffix:m/s\u00b2") var valid_int_0 : int
@export_custom(PROPERTY_HINT_NONE, "suffix:m/s\u00b2") var valid_int_1 = 0
@export_custom(PROPERTY_HINT_NONE, "suffix:m/s\u00b2") var valid_int_2 := 0
```
Godot_v4 3-stable_mono_win64_RIvv12JRU7

**Comment by tetrapod00** [comment]
You may also want to mention that vector types are supported. No need to list every type, but at least `float` and `Vector3`, and maybe also `int` would be good. Or you could just say "scalar and vector types" maybe.

Quick testing on which types can be used with this property hint:
```
# Unsupoorted
@export_custom(PROPERTY_HINT_NONE, "suffix:m/s\u00b2") var unsupported_bool : bool
@export_custom(PROPERTY_HINT_NONE, "suffix:m/s\u00b2") var unsupported_color : Color
@export_custom(PROPERTY_HINT_NONE, "suffix:m/s\u00b2") var unsupported_string : String
# Supported
@export_custom(PROPERTY_HINT_NONE, "suffix:m/s\u00b2") var supported_float : float
@export_custom(PROPERTY_HINT_NONE, "suffix:m/s\u00b2") var supported_int : int
@export_custom(PROPERTY_HINT_NONE, "suffix:m/s\u00b2") var supported_vec2 : Vector2
@export_custom(PROPERTY_HINT_NONE, "suffix:m/s\u00b2") var supported_vec2I : Vector2i
@export_custom(PROPERTY_HINT_NONE, "suffix:m/s\u00b2") var supported_vec3 : Vector3
@export_custom(PROPERTY_HINT_NONE, "suffix:m/s\u00b2") var supported_vec3I : Vector3i
@export_custom(PROPERTY_HINT_NONE, "suffix:m/s\u00b2") var supported_vec4 : Vector4
@export_custom(PROPERTY_HINT_NONE, "suffix:m/s\u00b2") var supported_vec4I : Vector4i
@export_custom(PROPERTY_HINT_NONE, "suffix:m/s\u00b2") var supported_quaternion : Quaternion
@export_custom(PROPERTY_HINT_NONE, "suffix:m/s\u00b2") var supported_basis : Basis
@export_custom(PROPERTY_HINT_NONE, "suffix:m/s\u00b2") var supported_rect2 : Rect2
# Partially supported
@export_custom(PROPERTY_HINT_NONE, "suffix:m/s\u00b2") var partial_transform2d : Transform2D
@export_custom(PROPERTY_HINT_NONE, "suffix:m/s\u00b2") var partial_transform3d : Transform3D

```
Godot_v4 3-stable_mono_win64_iWBYinG6ku

Some of these have nonsensical meanings - for example the transforms or quaternions. Some of these behaviors should maybe be changed on the engine side.

File: tutorials/scripting/c_sharp/c_sharp_exports.rst
Comment (tetrapod00) [inline tutorials/scripting/c_sharp/c_sharp_exports.rst:273]:
```suggestion
Display a unit hint suffix for exported variables. Works with numeric types, such as floats or vectors:
```

File: tutorials/scripting/c_sharp/c_sharp_exports.rst
Comment (tetrapod00) [inline tutorials/scripting/c_sharp/c_sharp_exports.rst:278]:
```suggestion
    public float Gravity { get; set; } = 9.8f;
    [Export(PropertyHint.None,"suffix:m/s")]
    public Vector3 Velocity { get; set; }
```
Public properties are used in the rest of the page. Also adds a vector example.

File: tutorials/scripting/gdscript/gdscript_exports.rst
Comment (tetrapod00) [inline tutorials/scripting/gdscript/gdscript_exports.rst:184]:
```suggestion
Display a unit hint suffix for exported variables. Works with numeric types, such as floats or vectors:
```

**Comment by ShawnHardern** [comment]
@tetrapod00 Thank you for your awesome suggestions 👍🏻 and for taking the time to test the values! 🙌🏻. I've gone ahead and added your suggestions🎉.

File: tutorials/scripting/c_sharp/c_sharp_exports.rst
Comment (tetrapod00) [inline tutorials/scripting/c_sharp/c_sharp_exports.rst:277]: Looking at this again, using `\u00b2` for the squared symbol may merit a comment or a mention in the preceding sentence. Personally I think it's fine as is - curiosity is a good teacher.

Other than that, LGTM but my approval as just a contributor is not the one that matters for getting these changes merged.

Comment (ShawnHardern) [reply tutorials/scripting/c_sharp/c_sharp_exports.rst:277]: Hey @tetrapod00 thanks for the feedback!

I agree that it's probably a good idea to add a comment / clarification regarding the use of `\u00b2`, I'll wait for input from maintainers to see how they'd like to proceed.

**Comment by Calinou** [comment]
- Superseded by https://github.com/godotengine/godot-docs/pull/9993. Thanks for the contribution nonetheless!

**Pull request closed without merging**
